core/capabilities/integration_tests/streams_test.go

```diff
@@ -0,0 +1,91 @@
+package integration_tests
+
+import (
+	"context"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/datastreams"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/keystone/generated/feeds_consumer"
+	"github.com/smartcontractkit/chainlink/v2/core/internal/testutils"
+)
+
+var (
+	workflowName    = "abcdef0123"
+	workflowOwnerID = "0100000000000000000000000000000000000001"
+)
+
+func Test_AllAtOnceTransmissionSchedule(t *testing.T) {
+	ctx := testutils.Context(t)
+
+	// The don IDs set in the below calls are inferred from the order in which the dons are added to the capabilities registry
+	// in the setupCapabilitiesRegistryContract function, should this order change the don IDs will need updating.
+	workflowDonInfo := createDonInfo(t, 1, 5, 1)
+	triggerDonInfo := createDonInfo(t, 2, 7, 1)
+	targetDonInfo := createDonInfo(t, 3, 4, 1)
+
+	consumer, feedIDs, triggerSink := setupStreamDonsWithTransmissionSchedule(ctx, t, workflowDonInfo, triggerDonInfo, targetDonInfo, 3,
+		"2s", "allAtOnce")
+
+	reports := []*datastreams.FeedReport{
+		createFeedReport(t, big.NewInt(1), 5, feedIDs[0], triggerDonInfo.keyBundles),
+		createFeedReport(t, big.NewInt(3), 7, feedIDs[1], triggerDonInfo.keyBundles),
+		createFeedReport(t, big.NewInt(2), 6, feedIDs[2], triggerDonInfo.keyBundles),
+	}
+
+	triggerSink.sendReports(reports)
+
+	waitForConsumerReports(t, consumer, reports)
+}
+
+func Test_OneAtATimeTransmissionSchedule(t *testing.T) {
+	ctx := testutils.Context(t)
+
+	// The don IDs set in the below calls are inferred from the order in which the dons are added to the capabilities registry
+	// in the setupCapabilitiesRegistryContract function, should this order change the don IDs will need updating.
+	workflowDonInfo := createDonInfo(t, 1, 5, 1)
+	triggerDonInfo := createDonInfo(t, 2, 7, 1)
+	targetDonInfo := createDonInfo(t, 3, 4, 1)
+
+	consumer, feedIDs, triggerSink := setupStreamDonsWithTransmissionSchedule(ctx, t, workflowDonInfo, triggerDonInfo, targetDonInfo, 3,
+		"2s", "oneAtATime")
+
+	reports := []*datastreams.FeedReport{
+		createFeedReport(t, big.NewInt(1), 5, feedIDs[0], triggerDonInfo.keyBundles),
+		createFeedReport(t, big.NewInt(3), 7, feedIDs[1], triggerDonInfo.keyBundles),
+		createFeedReport(t, big.NewInt(2), 6, feedIDs[2], triggerDonInfo.keyBundles),
+	}
+
+	triggerSink.sendReports(reports)
+
+	waitForConsumerReports(t, consumer, reports)
+}
+
+func setupStreamDonsWithTransmissionSchedule(ctx context.Context, t *testing.T, workflowDonInfo donInfo, triggerDonInfo donInfo, targetDonInfo donInfo,
```

Comment:
It doesn't bypass the syncer, that is being used as it would during normal run to setup the capabilities.  Perhaps you mean the transmission schedule?  Pre your recent changes, this is/was being passed in on the workflow, which is what's happening here.

Reply:
The syncer is being used to instantiate all capabilities apart from the trigger capability as the test needs a handle on this to inject the test messages.